docs2/site/docs/migrations/migration8.md

```diff
@@ -313,7 +314,28 @@ At this time GraphQL.NET does not directly support the `MaxLength` and similar a
 implement your own attributes as shown above, or call the `Validate` method to set a validation
 function.
 
-### 6. `@pattern` custom directive added for validating input values against a regular expression pattern
+### 6. `ArgumentValidator` delegate added to input object/interface field definitions (since 8.1)
+
+This allows for custom validation of the coerced argument values. It can be used when you must
+enforce certain relationships between arguments, such as ensuring that at least one of multiple
+arguments are provided, or that a specific argument is provided when another argument is set to
+a specific value. It can be set by configuring the `ValidateArguments` delegate on the `FieldType`
+class or calling the `ValidateArguments` method on the field builder. Here is an example:
+
+```csharp
+Field<string>("example")
+    .Argument<string>("str1", true)
+    .Argument<string>("str2", true)
+    .ValidateArguments(ctx =>
+    {
+        var str1 = ctx.GetArgument<string>("str1");
+        var str2 = ctx.GetArgument<string>("str2");
+        if (str1 == null && str2 == null)
+            throw new InvalidOperationException("Must provide str1 or str2");
+    });
```

Comment:
Execution is performed in stages, generally as follows:
1. Parse query document
2. Run pre-coercion validation rules
3. Coerce field and directive arguments
4. Run post-coercion validation rules
5. Execute document

Scalar methods, `Parser`, `Validator`, and `ParseDictionary` execute during stage 3.  That is why they do not have a context; they are simply coercing literals or variable values directly to internal mappings.

This PR adds a hook into stage 4.  During stage 4 it is not possible to use this technique on an input object type, as it may have already been nested a few levels deep under a field argument.  Only field arguments and directive arguments can be examined, much the same way that a field resolver only has access to the field and directive arguments.

That being said, the `InputObjectGraphType.ParseDictionary` method can perform some validation, being that it can inspect the values passed to it from its fields (post-coercion) and throw an exception if a validation rule on the input object as a whole has been violated.  For example, an input object with "FirstName", "LastName", and "FullName" fields could throw an exception within `ParseDictionary` if none of the three fields were supplied.